resources/js/components/PersonAvatar.vue

```diff
@@ -0,0 +1,49 @@
+<template>
+  <div ref="target">
+    <img v-if="targetWasVisibleOnce" :src="imageUrl" class="w-full rounded-lg" loading="lazy" />
+    <div v-else class="w-full rounded-lg h-52 bg-gray-900 animate-pulse" ></div>
+  </div>
+</template>
+
+<script>
+import { defineComponent, ref, watch } from "vue";
+import { useElementVisibility } from "@vueuse/core";
+
+export default defineComponent({
+  name: "PersonAvatar",
+  props: {
+    person: {
+      type: Object,
+      required: true
+    }
+  },
+  setup(props) {
+    const target = ref(null);
+    const targetIsVisible = useElementVisibility(target);
+    const targetWasVisibleOnce = ref(false);
+    const imageUrl = ref(null);
```

Comment:
Although it is not currently done this way in other places, it is cleaner to initialise with ‘undefined’ instead of ‘zero’. Say with `ref();` instead of `ref(null);`.